lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -0,0 +1,158 @@
+    match METRIC_SENDER.get() {
+        Some(storage) => {
+            let guard = storage.lock().unwrap();
+            guard
+                .try_send(metric)
+                .map_err(|e| generic_error!("Failed to send metric: {}", e))
+        }
+        None => Err(generic_error!("Metric sender not initialized.")),
+    }
+}
+
+#[pymodule]
+pub mod aggregator {
+    use super::*;
+
+    #[pymodule_export]
+    const GAUGE: i32 = MetricType::Gauge as i32;
+    #[pymodule_export]
+    const RATE: i32 = MetricType::Rate as i32;
+    #[pymodule_export]
+    const COUNT: i32 = MetricType::Count as i32;
+    #[pymodule_export]
+    const MONOTONIC_COUNT: i32 = MetricType::MonotonicCount as i32;
+    #[pymodule_export]
+    const COUNTER: i32 = MetricType::Counter as i32;
+    #[pymodule_export]
+    const HISTOGRAM: i32 = MetricType::Histogram as i32;
+    #[pymodule_export]
+    const HISTORATE: i32 = MetricType::Historate as i32;
+
+    #[allow(clippy::too_many_arguments)]
+    #[pyfunction]
+    fn submit_metric(
+        _class: PyObject, _check_id: String, mtype: i32, name: String, value: f64, tags: Vec<String>, hostname: String,
+        _flush_first_value: bool,
+    ) {
+        trace!(
+            "submit_metric called with name: {}, value: {}, tags: {:?}, hostname: {}",
+            name,
+            value,
+            tags,
+            hostname
+        );
+
+        let check_metric = CheckMetric::new(name.clone(), mtype.into(), value, tags.clone());
+
+        match try_send_metric(check_metric) {
+            Ok(()) => {}
+            Err(e) => {
+                error!("Failed to send metric: {}", e);
+            }
+        };
+    }
+
+    #[pyfunction]
+    #[pyo3(signature = (name, status, tags, hostname, message=None))]
+    fn submit_service_check(name: String, status: i32, tags: Vec<String>, hostname: String, message: Option<String>) {
+        println!(
+            "submit_service_check called with name: {}, status: {}, tags: {:?}, hostname: {}, message: {:?}",
+            name, status, tags, hostname, message
+        );
+    }
+
+    #[pyfunction]
+    fn metrics(name: String) -> Vec<String> {
+        println!("metrics called for: {}", name);
+        vec![] // Dummy return
+    }
+
+    #[pyfunction]
+    fn reset() {
+        println!("reset called");
+    }
+}
+
+#[pymodule]
+pub mod datadog_agent {
+    use super::*;
+
+    #[pyfunction]
+    fn get_hostname() -> &'static str {
+        trace!("Called get_hostname()");
+        "stubbed.hostname"
+    }
+
+    #[pyfunction]
+    fn set_hostname(hostname: String) {
+        trace!("Called set_hostname({})", hostname);
+        // In a function context without a struct, we cannot actually "set" the hostname persistently.
+    }
+
+    #[pyfunction]
+    fn reset_hostname() {
+        trace!("Called reset_hostname()");
+        // Similar to `set_hostname`, we cannot reset without a persistent structure.
+    }
```

Comment:
We let checks set and reset the hostname.... what the hell? 😂 

How does this checks-specific hostname actually get utilized?

Reply:
I copied these methods from @scottopell work during innovation week. (Shame on me for not double checking)

I believe these methods are not part of the building we offer to the Python checks. Doing a bit of search, I see we have them as part of the `DatadiogAegntStub ' in the integrations-core repo, probably for testing. 

https://github.com/DataDog/integrations-core/blob/98186cfdf5231a4efab3fbd05cae8f105da5430c/datadog_checks_base/datadog_checks/base/stubs/datadog_agent.py#L94-L98

I will remove them. 

> We let checks set and reset the hostname.... what the hell? 😂
That would be crazy if we allow Python to mess up with the hostname